sdk/core/azure-core/inc/azure/core/http/websockets/websockets_transport.hpp

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief Utilities to be used by HTTP WebSocket transport implementations.
+ */
+
+#pragma once
+
+#include "azure/core/context.hpp"
+#include "azure/core/http/http.hpp"
+
+namespace Azure { namespace Core { namespace Http { namespace WebSockets {
+
+  /**
+   * @brief Base class for all WebSocket transport implementations.
+   */
+  class WebSocketTransport {
+  public:
+    /**
+     * @brief Web Socket Frame type, one of Text or Binary.
+     */
+    enum class NativeWebSocketFrameType
+    {
+      /**
+       * @brief Indicates that the frame is a partial UTF-8 encoded text frame - it is NOT the
+       * complete frame to be sent to the remote node.
+       */
+      FrameTypeTextFragment,
```

Comment:
In code I would typically write 

`NativeWebSocketFrameType frametype = NativeWebSocketFrameType::FrameTypeTextFragment`

Seems wordy to start each value with FrameType vs:
`NativeWebSocketFrameType frametype = NativeWebSocketFrameType::TextFragment`

Reply:
Absolutely. That's just busted.